### PR TITLE
Add support for switch-case in mani

### PR DESCRIPTION
**Pull request by ankitshubham97**
---
name: Switch-case
about: I have added the feature of switch-case to the language.
---

**Is your PR related to a feature request or Bug report?**
If applicable, please list feature request number or bug report ID.
> It is related to an improvement feature #79.

**Describe your PR**
A clear and concise description of what your pull request is changing or adding.
> It is adding the feature of switch statements.

**Describe intended use**
A clear and concise description of the intended use of the feature. Along with example code.
> From #79: The addition of switch statements would make this language just that little bit better. It would remove the need for mass amounts of if statements. Sample code:
```
let a = 1;
switch (a) {
	case 1 :
		say "a";
	case 2 :
		say "b";
	case 3 :
		say "c";
}
```

**How did you fix the bug?**
If applicable, how did you fix the bug? Please use code snippets too.
> N/A

**Is it in the form of a library**
 - No
 
If applicable what is the library called?
> N/A

**Does your PR replace an existing system**
If applicable, please describe how this PR changes the use of a related item.
> No.

**Additional comments**
Limitations:

- ~~Does not support "default" keyword for now, added a TODO.~~
- ~~Does not support multiple statements within a case, only one statement per case works for now. So, something like following does not work:~~
```
switch (a) {
  case 1 :
    say "1";
    say "2";
}
```

### CODE REVIEW

**Comment by crazywolf132** [comment]
Great work. I am just going to hold off merging until it supports more than one line per case.

I will add a WIP tag to the title.

**Comment by ankitshubham97** [comment]
@crazywolf132 Commit 1625e28 now takes care of supporting more than one line per case.

**Comment by ankitshubham97** [comment]
@crazywolf132 I have a query. In src/com/mani/lang/core/Interpreter.java,: 
```
  @Override
  public Void visitBreakStmt(Stmt.Break stmt) {
    throw new Return();
  }
```
Why are we throwing a RuntimeException for 'break' statements? Is it because it is still unimplemented?

**Comment by crazywolf132** [comment]
@ankitshubham97  That was just how i chose to take care of break statements. If you believe there is a better way, by all means. Feel free to change it.

If you think this is good to go, please create a test case for it. Then we can merge.

File: src/com/mani/lang/core/Parser.java [hunk 1]
```diff
@@ -270,13 +272,22 @@ private Stmt switchStatement() {
         consume(TokenType.LEFT_BRACE, "Expect '{' after switch condition");
         while (! match(TokenType.RIGHT_BRACE)) {
             if (match(TokenType.CASE)) {
+                if (seenDefault) {
+                    throw new Return("Default is already declared before.");
```
Comment (crazywolf132) [inline src/com/mani/lang/core/Parser.java]: Please change this to `Default has already been declared.`

**Comment by crazywolf132** [comment]
@Kalekdan Anything to add?

File: src/com/mani/lang/core/Parser.java (hunk 1 above)
Comment (ankitshubham97) [reply src/com/mani/lang/core/Parser.java]: @crazywolf132 Addressed in c098248

**Comment by ankitshubham97** [comment]
@crazywolf132 Added test in c098248.

**Comment by crazywolf132** [comment]
Great work

**Comment by ankitshubham97** [comment]
Thanks! #79 can be closed now.

**Pull request merged**